kasa/tests/test_protocol.py

```diff
@@ -227,3 +249,63 @@ def test_decrypt_unicode():
     d = "{'snowman': '\u2603'}"
 
     assert d == TPLinkSmartHomeProtocol.decrypt(e)
+
+
+def _get_subclasses(of_class):
+    import kasa
+
+    package = sys.modules["kasa"]
+    subclasses = set()
+    for _, modname, _ in pkgutil.iter_modules(package.__path__):
+        importlib.import_module("." + modname, package="kasa")
+        module = sys.modules["kasa." + modname]
+        for name, obj in inspect.getmembers(module):
+            if inspect.isclass(obj) and issubclass(obj, of_class):
+                subclasses.add((name, obj))
+    return subclasses
+
+
+@pytest.mark.parametrize(
+    "class_name_obj", _get_subclasses(TPLinkProtocol), ids=lambda t: t[0]
+)
+def test_protocol_init_signature(class_name_obj):
+    params = list(inspect.signature(class_name_obj[1].__init__).parameters.values())
+
+    assert len(params) == 3
+    assert (
+        params[0].name == "self"
+        and params[0].kind == inspect.Parameter.POSITIONAL_OR_KEYWORD
+    )
+    assert (
+        params[1].name == "host"
+        and params[1].kind == inspect.Parameter.POSITIONAL_OR_KEYWORD
+    )
+    assert (
+        params[2].name == "transport"
+        and params[2].kind == inspect.Parameter.KEYWORD_ONLY
+    )
+
```

Comment:
I think it would have been enough here to have a static list of protocols & transports, and to try constructing them with some dummy values. But we can keep it like this, too.

Reply:
Ok cool